Source/monster.cpp

```diff
@@ -2713,19 +2669,15 @@ void SneakAi(int monsterId)
 	if (monster.mode != MonsterMode::Stand) {
 		return;
 	}
-	int mx = monster.position.tile.x;
-	int my = monster.position.tile.y;
-	if (dLight[mx][my] == LightsMax) {
+	if (dLight[monster.position.tile.x][monster.position.tile.y] == LightsMax) {
 		return;
 	}
-	mx -= monster.enemyPosition.x;
-	my -= monster.enemyPosition.y;
 
 	int dist = 5 - monster.intelligence;
 	if (static_cast<MonsterMode>(monster.var1) == MonsterMode::HitRecovery) {
 		monster.goal = MonsterGoal::Retreat;
 		monster.goalVar1 = 0;
-	} else if (abs(mx) >= dist + 3 || abs(my) >= dist + 3 || monster.goalVar1 > 8) {
+	} else if (monster.longestDistanceToEnemy() >= dist + 3 || monster.goalVar1 > 8) {

```

Comment:
Looks like SneakAi() makes multiple uses of longestDistanceToEnemy(), pull it in to a variable to avoid running the logic multiple times.